### DISCUSSION TITLE
Support for OpenSearch Indexing backend

### DISCUSSION THREAD

**Discussion by langrp** [JanusGraph/janusgraph#2837] [Ideas]
Hi, We'd like to be able to use OpenSearch with JanusGraph. Unfortunately the OpenSearch project uses its own versioning and therefore is unable to be used instead of ElasticSearch indexing backend.

I believe this could be simply change as OpenSearch clients are not yet build and officially it uses ElasticSearch clients.

Any thoughts?
Thanks

**Comment by stockholmux** [comment]
Hey @langrp - if this is truly a version issue, you can have OpenSearch identify itself as 7.10.2 using `compatibility.override_main_response_version: true` in your `opensearch.yml` (it's [documented](https://opensearch.org/docs/latest/clients/agents-and-ingestion-tools/index/), but in a weird place)

**Reply by langrp** [reply]
Yeah, thanks a bunch @stockholmux , I did not notice this compatibility mode of OpenSearch, very strange place. I've tried that and so far so good.

**Reply by Alvaro360** [reply]
Hi, does it work? I'm trying to use OpenSearch with JanusGraph as well.

**Reply by Phate334** [reply]
It is work, this is my docker compose file. 
https://gist.github.com/Phate334/9fd90ddce0eaea74086c1a2e5d419402

**Reply by Alvaro360** [reply]
Worked for me! thanks a lot.

**Reply by stockholmux** [reply]
Just an FYI - that option will likely go away in OpenSearch 2.0.0 as there are breaking API changes vs ES 7.10.2 (minor, but existent). I'd love to see JanusGraph support OpenSearch properly.

**Reply by yingjie668** [reply]
> It is work, this is my docker compose file. https://gist.github.com/Phate334/9fd90ddce0eaea74086c1a2e5d419402

Hello, when I tried docker-compose --verbose up, I can see all four containers are up and running. But there are error such as out memory (cassandra/opensearch), connection refuse error for opensearch-dashboards . Here is the full trace.  So  how to confirm that these containers are connected and can communicate with each other?  

[docker_compose_verbose.txt](https://github.com/JanusGraph/janusgraph/files/10941000/docker_compose_verbose.txt)
